docs/language-agnostic-definition.md

```diff
@@ -0,0 +1,99 @@
+An Autonomous Economic Agent is, in technical terms, defined by the following characteristics:
+
+- It MUST be capable of receiving and sending `Envelopes` which satisfy the following protobuf schema:
+
+``` proto
+syntax = "proto3";
+
+package fetch.aea;
+
+message Envelope{
+    string to = 1;
+    string sender = 2;
+    string protocol_id = 3;
+    bytes message = 4;
+    string uri = 5;
+}
+```
+
+<div class="admonition note">
+  <p class="admonition-title">Note</p>
+  <p>Additional constraints on envelope fields will be added soon!</p>
+</div>
+
+- It MUST implement each protocol with the required meta-fields:
+
+``` proto
+
+    // Meta fields
+    int32 message_id = 1;
+    string dialogue_starter_reference = 2;
+    string dialogue_responder_reference = 3;
+    int32 target = 4;
+    oneof performative{
+        ...
+    }
+```
+ where `...` is replaced with the protocol specific performatives.
+
+- It MUST implement protocols according to their specification.
+
+<div class="admonition note">
+  <p class="admonition-title">Note</p>
+  <p>This section is incomplete, and will be updated soon!</p>
+</div>
+
+- It SHOULD implement the `fetchai/default:0.1.0` protocol which satisfies the following protobuf schema:
+
+``` proto
+syntax = "proto3";
+
+package fetch.aea.Default;
+
+message DefaultMessage{
+
+    // Custom Types
+    message ErrorCode{
+        enum ErrorCodeEnum {
+            UNSUPPORTED_PROTOCOL = 0;
+            DECODING_ERROR = 1;
+            INVALID_MESSAGE = 2;
+            UNSUPPORTED_SKILL = 3;
+            INVALID_DIALOGUE = 4;
+          }
+        ErrorCodeEnum error_code = 1;
+    }
+
+
+    // Performatives and contents
+    message Bytes_Performative{
+        bytes content = 1;
+    }
+
+    message Error_Performative{
+        ErrorCode error_code = 1;
+        string error_msg = 2;
+        map<string, bytes> error_data = 3;
+    }
+
+
+    // Standard DefaultMessage fields
+    int32 message_id = 1;
+    string dialogue_starter_reference = 2;
+    string dialogue_responder_reference = 3;
+    int32 target = 4;
+    oneof performative{
+        Bytes_Performative bytes = 5;
+        Error_Performative error = 6;
+    }
+}
+```
+
+- It MUST process `Envelopes` asynchronously.
```

Comment:
Don't agree on this. I'd say `SHOULD`

Reply:
Actually, I wouldn't write this at all. I see the specifications as a definition of an interface; the others don't need to know about the implementation to interact with an AEA, it just have to comply with the interface.

Reply:
What you might want to add is a sort of [non-functional requirement](https://en.wikipedia.org/wiki/Non-functional_requirement), which could go roughly as follows:

> The specification regarding the processing of messages does not impose any particular implementation choice/constraint; for example, the AEA can process envelopes either synchronously and asynchronously. However, due to the high level of activity that an AEA might be subject to, others AEAs expect a certain minimum level of responsiveness and reactivity of an AEA's implementation, especially in the case of many concurrent dialogues with other peers. That could imply the need for asynchronous programming to make the AEA's implementation scalable.

Reply:
Let's add the non-functional requirement